### PR TITLE
Rewrite args parsing to use cliffy

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Towards #20.

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
The fix in 36f42eefa11b3def055cdfac690afba99ae862ed uses `getLocals` which is documented as brittle. However, I did not find another way to obtain variable declarations from object binding patters such as `const { Deno } = globalThis;`. I trust E2E tests enough that this should work in the long run.

In addition, the declared locals include type declarations, not only variables. Is this desired? What if we declare an interface, and then shim a function with the same name? I think we then skip the shimming, however, those two declarations do not actually clash. Can this cause a problem?

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
     "node": ">=14.13.1"
   },
   "dependencies": {
-    "ts-morph": "^13.0.2",
+    "deno.ns": "^0.8.0",
```
Comment (wojpawlik) [inline package.json:44]: `deno.ns` is deprecated in favor of https://www.npmjs.com/package/@deno/shim-deno

**Review by wojpawlik — APPROVED** [review]
> What if we declare an interface, and then shim a function with the same name?

TypeScript will report a conflicting definition, same as before this PR.

**Comment by KnorpelSenf** [comment]
> > What if we declare an interface, and then shim a function with the same name?
> 
> TypeScript will report a conflicting definition, same as before this PR.

Did you test this? Because I am able to
```ts
interface A {}
const A = 0
```
and the code compiles.

File: package.json (hunk 1 above)
Comment (KnorpelSenf) [reply package.json:44]: Alright, I'll change it.

**Comment by wojpawlik** [comment]
Yes I did: https://github.com/microsoft/TypeScript/issues/37712

**Comment by KnorpelSenf** [comment]
Understood. Merging.

**Pull request merged**